File: Data Structures & Algorithms/duplicate-integer/submission-4.cpp
class Solution {
public:
    bool hasDuplicate(vector<int>& nums) {
        const int size = nums.size();
        int currentNum;
        for (int i = 0; i < size; i++) {
            currentNum = nums.at(i);
            for (int j = i+1; j < size; j++) {
                if (nums.at(j) == currentNum) {
                    return true;
                }
            }
        }
        return false;
    }
}

File: Data Structures & Algorithms/contains-duplicate-ii/submission-4.py
class Solution:
    def containsNearbyDuplicate(self, nums: list[int], k: int) -> bool:
        left = 0
        right = 1
        while right <= len(nums):
            while abs(left - right) <= k:
                section = nums[left : right + 1]
                if len(set(section)) < len(section):
                    return True
                right += 1
            left += 1
            right = left + 1
        return False